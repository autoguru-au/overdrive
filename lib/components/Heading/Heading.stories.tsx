import { Meta, StoryObj } from '@storybook/react';
import React, { type ComponentProps } from 'react';

import { Text } from '../Text';

import { Heading } from '.';

const noWrapOptions: Array<ComponentProps<typeof Heading>['noWrap']> = [
	false,
	true,
];
const wordBreakOptions: Array<ComponentProps<typeof Text>['wordBreak']> = [
	false,
	true,
];
const transformOptions: Array<ComponentProps<typeof Text>['transform']> = [
	'uppercase',
	'capitalize',
	undefined,
];
const weightOptions: Array<ComponentProps<typeof Text>['fontWeight']> = [
	'normal',
	'semiBold',
	'bold',
];
const headingTypeOptions: Array<ComponentProps<typeof Heading>['is']> = [
	'h1',
	'h2',
	'h3',
	'h4',
	'h5',
	'h6',
];
const sizeScale: Array<ComponentProps<typeof Heading>['size']> = [
	'1',
	'2',
	'3',
	'4',
	'5',
	'6',
	'7',
	'8',
	'9',
];
const alignOptions: Array<ComponentProps<typeof Heading>['align']> = [
	'left',
	'center',
	'right',
];
const colourOptions: Array<ComponentProps<typeof Heading>['colour']> = [
	'dark',
	'light',
	'neutral',
	'primary',
	'secondary',
	'white',
	'information',
	'link',
	'success',
	'danger',
	'warning',
	'shine',
];

export default {
	title: 'Primatives/Heading',
	component: Heading, //Breaks the docs when enabled!
	argTypes: {
		noWrap: {
			options: noWrapOptions,
			defaultValue: false,
			control: {
				type: 'boolean',
			},
		},
		breakWord: {
			options: wordBreakOptions,
			defaultValue: false,
			control: {
				type: 'boolean',
			},
		},
		transform: {
			options: transformOptions,
			defaultValue: null,
			control: {
				type: 'select',
			},
		},
		weight: {
			options: weightOptions,
			defaultValue: null,
			control: {
				type: 'select',
			},
		},
		size: {
			options: sizeScale,
			defaultValue: void 0,
			control: {
				type: 'select',
			},
		},
		align: {
			options: alignOptions,
			defaultValue: 'left',
			control: {
				type: 'select',
			},
		},
		colour: {
			options: colourOptions,
			defaultValue: void 0,
			control: {
				type: 'select',
			},
		},
		is: {
			options: headingTypeOptions,
			defaultValue: 'h1',
			table: {
				type: { summary: 'heading' },
				defaultValue: 'h1',
			},
			description: 'HTML heading tag to be used',
			control: {
				type: 'select',
			},
		},
	},
} satisfies Meta<typeof Heading>;

type Story = StoryObj<typeof Heading>;

export const Standard: Story = {
	args: {
		is: 'h1',
		children: 'I am a heading',
	},
	render: (args) => (
		<div style={{ maxWidth: '350px', width: '100%' }}>
			<Heading {...args} />
		</div>
	),
};

export const AllTypes: Story = {
	args: {
		children: 'I am a heading',
	},
	render: (args) => (
		<div style={{ maxWidth: '350px', width: '100%' }}>
			{headingTypeOptions.map((is) => (
				<Heading key={is} {...args} is={is} />
			))}
		</div>
	),
};

export const AllColours: Story = {
	args: {
		children: 'I am a heading',
	},
	render: (args) => (
		<div style={{ maxWidth: '350px', width: '100%' }}>
			{colourOptions.map((colour, index) => (
				<div key={index} style={{ marginBottom: 8 }}>
					<Heading {...args} colour={colour} />
				</div>
			))}
		</div>
	),
};
