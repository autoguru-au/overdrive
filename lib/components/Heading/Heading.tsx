import React from 'react';

import type { Sprinkles } from '../../styles/sprinkles.css';
import { Box, type BoxProps } from '../Box';
import { textStyles, type TextStylesProps } from '../Text/textStyles';

export type HeadingTags = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';

export interface HeadingProps
	extends Omit<BoxProps, keyof TextStylesProps>,
		Omit<TextStylesProps, 'as'> {
	as?: HeadingTags;
	align?: TextStylesProps['textAlign'];
	/** @deprecated Prefer `as` */
	is?: HeadingTags;
}

const defaultSizeMap: Record<HeadingTags, Sprinkles['fontSize']> = {
	h1: '8',
	h2: '7',
	h3: '6',
	h4: '4',
	h5: '3',
	h6: '2',
};

/**
 * Heading renders an <h1... h6> with default font size and weight
 */
export const Heading = ({
	is = 'h1',
	as = is,
	align,
	breakWord,
	children,
	className,
	color, // semantic tokens
	colour = color ? undefined : 'dark', // legacy intentional tokens
	fontWeight = 'bold',
	noWrap,
	size = defaultSizeMap[as],
	transform,
	...props
}: HeadingProps) => (
	<Box
		{...props}
		as={as}
		color={color}
		className={[
			textStyles({
				breakWord,
				colour,
				fontWeight,
				noWrap,
				size,
				textAlign: align,
				transform,
			}),
			className,
		]}
	>
		{children}
	</Box>
);
