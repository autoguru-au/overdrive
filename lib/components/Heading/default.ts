export { Heading as default } from './Heading';
