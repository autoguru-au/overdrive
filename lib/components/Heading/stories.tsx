import { boolean, select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';

import { EHeadingSize, EHeadingVariant, Heading } from '.';
import { HeadingLoader } from './HeadingLoader';

const baseProps = () => ({
	variant: select('Variant', { Nothing: null, ...EHeadingVariant }, null),
});

const baseLoadingProps = () => ({
	size: select('Size', EHeadingSize, EHeadingSize.Heading1),
	randomWidth: boolean('randomWidth', true),
	blinking: boolean('blinking', true),
});

storiesOf('Components|Type/Heading', module)
	.add('heading1', () => (
		<Heading {...baseProps()} size={EHeadingSize.Heading1}>
			Hello World
		</Heading>
	))
	.add('heading2', () => (
		<Heading {...baseProps()} size={EHeadingSize.Heading2}>
			Hello World
		</Heading>
	))
	.add('heading3', () => (
		<Heading {...baseProps()} size={EHeadingSize.Heading3}>
			Hello World
		</Heading>
	))
	.add('heading4', () => (
		<Heading {...baseProps()} size={EHeadingSize.Heading4}>
			Hello World
		</Heading>
	))
	.add('heading5', () => (
		<Heading {...baseProps()} size={EHeadingSize.Heading5}>
			Hello World
		</Heading>
	))
	.add('heading6', () => (
		<Heading {...baseProps()} size={EHeadingSize.Heading6}>
			Hello World
		</Heading>
	))
	.add(
		'loading',
		() => <HeadingLoader {...baseProps()} {...baseLoadingProps()} />,
		{ chromatic: { disable: true } }
	);
