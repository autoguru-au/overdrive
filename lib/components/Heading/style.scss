@import '../../theme/tokens/facade';

@function calcTextLoaderMargin($textHeight, $lineHeight) {
	@return calc(0.5 * (#{$lineHeight} - #{$textHeight}));
}

:root {
	--global--heading--1--size-mobile: 30px;
	--global--heading--1--size-desktop: 32px;
	--global--heading--1--height-mobile: 36px;
	--global--heading--1--height-desktop: 40px;
	--global--heading--2--size-mobile: 26px;
	--global--heading--2--size-desktop: 28px;
	--global--heading--2--height-mobile: 30px;
	--global--heading--2--height-desktop: 36px;
	--global--heading--3--size-mobile: 22px;
	--global--heading--3--size-desktop: 24px;
	--global--heading--3--height-mobile: 28px;
	--global--heading--3--height-desktop: 32px;
	--global--heading--4--size-mobile: 18px;
	--global--heading--4--size-desktop: 20px;
	--global--heading--4--height-mobile: 24px;
	--global--heading--4--height-desktop: 28px;
	--global--heading--5--size: 16px;
	--global--heading--5--height: 22px;
	--global--heading--6--size: 14px;
	--global--heading--6--height: 18px;
}

@mixin variantDesktop() {
	&.variantDesktop {
		@content;
	}

	&:not(.variantMobile) {
		@include isDesktop() {
			// TODO update desktop query when rules are defined around screen sizes
			@content;
		}
	}
}

.root {
	padding: 0;
	margin: 0;
	font-weight: 700;
}

.Heading1 {
	font-size: var(--global--heading--1--size-mobile);
	line-height: var(--global--heading--1--height-mobile);

	@include variantDesktop() {
		line-height: var(--global--heading--1--height-desktop);
		font-size: var(--global--heading--1--size-desktop);
	}
}

.Heading2 {
	font-size: var(--global--heading--2--size-mobile);
	line-height: var(--global--heading--2--height-mobile);

	@include variantDesktop() {
		line-height: var(--global--heading--2--height-desktop);
		font-size: var(--global--heading--2--size-desktop);
	}
}

.Heading3 {
	font-size: var(--global--heading--3--size-mobile);
	line-height: var(--global--heading--3--height-mobile);

	@include variantDesktop() {
		line-height: var(--global--heading--3--height-desktop);
		font-size: var(--global--heading--3--size-desktop);
	}
}

.Heading4 {
	font-size: var(--global--heading--4--size-mobile);
	line-height: var(--global--heading--4--height-mobile);

	@include variantDesktop() {
		line-height: var(--global--heading--4--height-desktop);
		font-size: var(--global--heading--4--size-desktop);
	}
}

.Heading5 {
	font-size: var(--global--heading--5--size);
	line-height: var(--global--heading--5--height);
}

.Heading6 {
	font-size: var(--global--heading--6--size);
	line-height: var(--global--heading--6--height);
}

.loader {
	&-Heading1 {
		height: var(--global--heading--1--size-mobile);
		margin: calcTextLoaderMargin(
				var(--global--heading--1--size-mobile),
				var(--global--heading--1--height-mobile)
			)
			0;

		@include variantDesktop() {
			height: var(--global--heading--1--size-desktop);
			margin: calcTextLoaderMargin(
					var(--global--heading--1--size-desktop),
					var(--global--heading--1--height-desktop)
				)
				0;
		}
	}

	&-Heading2 {
		height: var(--global--heading--2--size-mobile);
		margin: calcTextLoaderMargin(
				var(--global--heading--2--size-mobile),
				var(--global--heading--2--height-mobile)
			)
			0;

		@include variantDesktop() {
			height: var(--global--heading--2--size-desktop);
			margin: calcTextLoaderMargin(
					var(--global--heading--2--size-desktop),
					var(--global--heading--2--height-desktop)
				)
				0;
		}
	}

	&-Heading3 {
		height: var(--global--heading--3--size-mobile);
		margin: calcTextLoaderMargin(
				var(--global--heading--3--size-mobile),
				var(--global--heading--3--height-mobile)
			)
			0;

		@include variantDesktop() {
			height: var(--global--heading--3--size-desktop);
			margin: calcTextLoaderMargin(
					var(--global--heading--3--size-desktop),
					var(--global--heading--3--height-desktop)
				)
				0;
		}
	}

	&-Heading4 {
		height: var(--global--heading--4--size-mobile);
		margin: calcTextLoaderMargin(
				var(--global--heading--4--size-mobile),
				var(--global--heading--4--height-mobile)
			)
			0;

		@include variantDesktop() {
			height: var(--global--heading--4--size-desktop);
			margin: calcTextLoaderMargin(
					var(--global--heading--4--size-desktop),
					var(--global--heading--4--height-desktop)
				)
				0;
		}
	}

	&-Heading5 {
		height: var(--global--heading--5--size);
		margin: calcTextLoaderMargin(
				var(--global--heading--5--size),
				var(--global--heading--5--height)
			)
			0;
	}

	&-Heading6 {
		height: var(--global--heading--6--size);
		margin: calcTextLoaderMargin(
				var(--global--heading--6--size),
				var(--global--heading--6--height)
			)
			0;
	}
}
