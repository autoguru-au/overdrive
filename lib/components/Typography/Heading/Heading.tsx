import cx from 'clsx';
import React, { FunctionComponent } from 'react';
import { TSizeScale } from '../types';
import styles from './style.scss';

export interface IProps {
	className?: string;
	is?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
	size?: TSizeScale;
}

const sizeScaleDefaults: Map<string, TSizeScale> = new Map([
	['h1', 6],
	['h2', 5],
	['h3', 4],
	['h4', 3],
	['h5', 2],
	['h6', 1],
]);

export const Heading: FunctionComponent<IProps> = ({
	children,
	className = '',
	is: Component = 'h1',
	size = sizeScaleDefaults.get(Component),
}) => (
	<Component
		className={cx([
			styles.root,
			{ [styles[`sizeScale${size}`]]: !!size },
			className,
		])}
		children={children}
	/>
);
