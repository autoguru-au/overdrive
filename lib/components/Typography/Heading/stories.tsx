import { select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';

import { Heading } from '.';

storiesOf('Foundation|Typography/Heading', module)
	.addDecorator(story => (
		<div style={{ maxWidth: '350px', width: '100%' }}>{story()}</div>
	))
	.add('default', () => (
		<Heading
			is={select('Tag', ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'], 'h1')}
			size={select('Size', [1, 2, 3, 4, 5, 6, 7, 8, 9])}>
			Hello World
		</Heading>
	));
