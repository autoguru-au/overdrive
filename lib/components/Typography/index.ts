export * from './Heading';
export * from './Text';
export * from './TextLink';
