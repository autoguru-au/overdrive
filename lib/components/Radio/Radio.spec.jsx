import React, { useState } from 'react';
import { Radio, RadioGroup } from '.';
import { mount, render } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { Heading } from '../Typography/Heading';

describe('<RadioButton />', () => {
	describe('when not nested in a RadioGroup', () => {
		it('should throw if a RadioButton element is not nested inside a RadioGroup', () =>
			expect(() => render(<Radio value="1" />)).toThrow());
	});

	describe('when nested in a RadioGroup', () => {
		it('should not throw', () =>
			expect(() =>
				mount(
					<RadioGroup name="radio">
						<Radio />
					</RadioGroup>,
				),
			).not.toThrow());

		it('should match the snapshot for a single radio with no value or label', () =>
			expect(
				mount(
					<RadioGroup name="radio">
						<Radio />
					</RadioGroup>,
				),
			).toMatchSnapshot());

		it('should match the snapshot for a single radio', () =>
			expect(
				mount(
					<RadioGroup name="radio">
						<Radio children="radio label 1" value="1" />
					</RadioGroup>,
				),
			).toMatchSnapshot());

		it('should match the snapshot for a checked radio', () =>
			expect(
				mount(
					<RadioGroup name="radio">
						<Radio children="radio label 1" checked value="1" />
						<Radio
							children="radio label 2"
							value="2"
							checked={false}
						/>
					</RadioGroup>,
				),
			).toMatchSnapshot());

		it('should match the snapshot for a focused radio', () =>
			expect(
				mount(
					<RadioGroup name="radio">
						<Radio children="radio label 1" value="1" />
					</RadioGroup>,
				)
					.find(Radio)
					.simulate('focus'),
			).toMatchSnapshot());

		it('should match the snapshot for a group of radios', () =>
			expect(
				render(
					<RadioGroup name="radio">
						<Radio children="radio label 1" value="1" />
						<Radio children="radio label 2" value="2" />
						<Radio children="radio label 3" value="3" />
						<Radio children="radio label 4" value="4" />
					</RadioGroup>,
				),
			).toMatchSnapshot());

		it('should automatically select the radio with value equal to the value of its radiogroup', () =>
			expect(
				render(
					<RadioGroup name="radio" value="2">
						<Radio children="radio label 1" value="1" />
						<Radio children="radio label 2" value="2" />
						<Radio children="radio label 3" value="3" />
						<Radio children="radio label 4" value="4" />
					</RadioGroup>,
				)
					.find('>div:nth-of-type(2)')
					.hasClass('checked'),
			).toBeTruthy());

		it('should select the radio after it has been clicked', () => {
			let group;
			act(() => {
				const TestComponent = () => {
					const [value, setValue] = useState();

					return (
						<RadioGroup
							name="radio"
							value={value}
							onChange={setValue}>
							<Radio children="radio label 1" value="1" />
							<Radio children="radio label 2" value="2" />
							<Radio children="radio label 3" value="3" />
							<Radio children="radio label 4" value="4" />
						</RadioGroup>
					);
				};

				group = mount(<TestComponent />);
			});

			expect(
				group
					.find('div.radio')
					.at(3)
					.hasClass('checked'),
			).not.toBeTruthy();

			act(() => {
				group
					.find("div.radio>input[type='radio']")
					.at(3)
					.simulate('click');
			});

			group.update();

			expect(
				group
					.find('div.radio')
					.at(3)
					.hasClass('checked'),
			).toBeTruthy();
		});

		it('should not throw is onChange callback is not attached', () => {
			let group;
			act(() => {
				const TestComponent = () => {
					return (
						<RadioGroup name="radio" value="2">
							<Radio children="radio label 1" value="1" />
							<Radio children="radio label 2" value="2" />
							<Radio children="radio label 3" value="3" />
							<Radio children="radio label 4" value="4" />
						</RadioGroup>
					);
				};

				group = mount(<TestComponent />);
			});

			expect(
				group
					.find('div.radio')
					.at(2)
					.hasClass('checked'),
			).not.toBeTruthy();

			act(() => {
				expect(() =>
					group
						.find("div.radio>input[type='radio']")
						.at(3)
						.simulate('click'),
				).not.toThrow();
			});
		});

		it('should select the radio if the native radio element has been clicked', () => {
			let group;
			act(() => {
				const TestComponent = () => {
					const [value, setValue] = useState();

					return (
						<RadioGroup
							name="radio"
							value={value}
							onChange={setValue}>
							<Radio children="radio label 1" value="1" />
							<Radio children="radio label 2" value="2" />
							<Radio children="radio label 3" value="3" />
							<Radio children="radio label 4" value="4" />
						</RadioGroup>
					);
				};

				group = mount(<TestComponent />);
			});

			expect(
				group
					.find('div.radio')
					.at(2)
					.hasClass('checked'),
			).not.toBeTruthy();

			act(() => {
				const radio = group.find("div.radio>input[type='radio']").at(2);

				radio.simulate('click');
			});

			group.update();
			expect(
				group
					.find('div.radio')
					.at(2)
					.hasClass('checked'),
			).toBeTruthy();
		});

		it('should call the onClick function passed down to the radio button when it has been clicked', () => {
			let group;
			const spyedChangeCallback = jest.fn();
			act(() => {
				group = mount(
					<RadioGroup
						name="radio"
						value="2"
						onChange={spyedChangeCallback}>
						<Radio children="radio label 1" value="1" />
						<Radio children="radio label 2" value="2" />
						<Radio children="radio label 3" value="3" />
						<Radio children="radio label 4" value="4" />
					</RadioGroup>,
				);
			});

			act(() => {
				group
					.find("div.radio>input[type='radio']")
					.at(1)
					.simulate('click');
			});

			group.update();

			expect(spyedChangeCallback).toHaveBeenCalledTimes(1);
		});

		it('should disable the native radio element if disabled prop is set to true', () => {
			let group;
			act(() => {
				group = mount(
					<RadioGroup name="radio" value="2">
						<Radio children="radio label 1" value="1" />
						<Radio children="radio label 2" disabled value="2" />
						<Radio children="radio label 3" value="3" />
						<Radio children="radio label 4" value="4" />
					</RadioGroup>,
				);
			});

			expect(
				group
					.find("div.radio>input[type='radio']")
					.at(1)
					.is('[disabled]'),
			).toBeTruthy();
		});
	});

	describe('when with component child', () => {
		it('should match the snapshot', () => {
			const radio = render(
				<RadioGroup name="radio" value="2">
					<Radio children="radio label 1" value="1">
						<Heading>Hello checkbox</Heading>
					</Radio>
				</RadioGroup>,
			);

			expect(radio.html()).toMatchSnapshot();
		});

		it('Should render the component child', () => {
			const radio = mount(
				<RadioGroup name="radio" value="2">
					<Radio children="radio label 1" value="1">
						<Heading>Hello checkbox</Heading>
					</Radio>
				</RadioGroup>,
			);

			expect(radio.find(Heading)).toBeTruthy();
		});
	});
});
