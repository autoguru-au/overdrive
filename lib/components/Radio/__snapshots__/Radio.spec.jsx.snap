// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> when nested in a RadioGroup should match the snapshot for a checked radio 1`] = `
<div
  class="radioGroup"
>
  <div
    class="root radio"
  >
    <input
      class="nativeInput"
      name="radio"
      type="radio"
      value="1"
    />
    <div
      class="checkable"
    >
      <div
        class="outerCircle"
      />
      <div
        class="innerCircle"
      />
    </div>
    <div
      class="focusRect"
    />
    <label
      class="label"
    >
      radio label 1
    </label>
  </div>
  <div
    class="root radio"
  >
    <input
      class="nativeInput"
      name="radio"
      type="radio"
      value="2"
    />
    <div
      class="checkable"
    >
      <div
        class="outerCircle"
      />
      <div
        class="innerCircle"
      />
    </div>
    <div
      class="focusRect"
    />
    <label
      class="label"
    >
      radio label 2
    </label>
  </div>
</div>
`;

exports[`<RadioButton /> when nested in a RadioGroup should match the snapshot for a group of radios 1`] = `
<div
  class="radioGroup"
>
  <div
    class="root radio"
  >
    <input
      class="nativeInput"
      name="radio"
      type="radio"
      value="1"
    />
    <div
      class="checkable"
    >
      <div
        class="outerCircle"
      />
      <div
        class="innerCircle"
      />
    </div>
    <div
      class="focusRect"
    />
    <label
      class="label"
    >
      radio label 1
    </label>
  </div>
  <div
    class="root radio"
  >
    <input
      class="nativeInput"
      name="radio"
      type="radio"
      value="2"
    />
    <div
      class="checkable"
    >
      <div
        class="outerCircle"
      />
      <div
        class="innerCircle"
      />
    </div>
    <div
      class="focusRect"
    />
    <label
      class="label"
    >
      radio label 2
    </label>
  </div>
  <div
    class="root radio"
  >
    <input
      class="nativeInput"
      name="radio"
      type="radio"
      value="3"
    />
    <div
      class="checkable"
    >
      <div
        class="outerCircle"
      />
      <div
        class="innerCircle"
      />
    </div>
    <div
      class="focusRect"
    />
    <label
      class="label"
    >
      radio label 3
    </label>
  </div>
  <div
    class="root radio"
  >
    <input
      class="nativeInput"
      name="radio"
      type="radio"
      value="4"
    />
    <div
      class="checkable"
    >
      <div
        class="outerCircle"
      />
      <div
        class="innerCircle"
      />
    </div>
    <div
      class="focusRect"
    />
    <label
      class="label"
    >
      radio label 4
    </label>
  </div>
</div>
`;

exports[`<RadioButton /> when nested in a RadioGroup should match the snapshot for a single radio 1`] = `
<div
  class="radioGroup"
>
  <div
    class="root radio"
  >
    <input
      class="nativeInput"
      name="radio"
      type="radio"
      value="1"
    />
    <div
      class="checkable"
    >
      <div
        class="outerCircle"
      />
      <div
        class="innerCircle"
      />
    </div>
    <div
      class="focusRect"
    />
    <label
      class="label"
    >
      radio label 1
    </label>
  </div>
</div>
`;

exports[`<RadioButton /> when nested in a RadioGroup should match the snapshot for a single radio with no value or label 1`] = `
<div
  class="radioGroup"
>
  <div
    class="root radio"
  >
    <input
      class="nativeInput"
      name="radio"
      type="radio"
      value=""
    />
    <div
      class="checkable"
    >
      <div
        class="outerCircle"
      />
      <div
        class="innerCircle"
      />
    </div>
    <div
      class="focusRect"
    />
    <label
      class="label"
    />
  </div>
</div>
`;

exports[`<RadioButton /> when with component child should match the snapshot 1`] = `
<div
  class="radioGroup"
>
  <div
    class="root radio"
  >
    <input
      class="nativeInput"
      name="radio"
      type="radio"
      value="1"
    />
    <div
      class="checkable"
    >
      <div
        class="outerCircle"
      />
      <div
        class="innerCircle"
      />
    </div>
    <div
      class="focusRect"
    />
    <label
      class="label"
    >
      <h1
        class="root sizeScale6"
      >
        Hello checkbox
      </h1>
    </label>
  </div>
</div>
`;

exports[`<RadioButton /> when with component child should render the component child 1`] = `
<h1
  class="root sizeScale6"
>
  Hello checkbox
</h1>
`;
